src/components/Channel/Channel.tsx

```diff
@@ -1194,9 +1215,13 @@ const ChannelWithContext = <
     Re,
     Us
   >['retrySendMessage'] = async (message) => {
-    message = { ...message, status: 'sending' };
-    updateMessage(message);
-    await sendMessageRequest(message);
+    const statusPendingMessage = {
+      ...message,
+      status: 'sending',
+    };
+
+    updateMessage(statusPendingMessage);
```

Comment:
I think we can change this to:

```
  const retrySendMessage: MessagesContextValue<
    At,
    Ch,
    Co,
    Ev,
    Me,
    Re,
    Us
  >['retrySendMessage'] = async (message) => {
    updateMessage({ ...message, status: 'sending' });
    await sendMessageRequest({ ...message, id: undefined });
  };
```

Then inside of `sendMessageRequest` you do not need to change `id: retrying ? undefined : id,`, it can just be `id,`. `status` is unused in `sendMessageRequest` so we don't need to add it for that call, and then instead we can set the `id` to undefined here in the input and not change the function itself.

Reply:
When we try to send a message and have no network connection, the message is added with a frontend-generated id. I still need to send the old message id when retrying because I want to use that id to replace the failed message with the new message (which would be created with a new backend-generated id because of `id: retrying ? undefined : id`:

```
if (retrying) {
  replaceMessage(message, messageResponse.message);
}
```

Before this pr, we would end up having 2 messages in the chat (both front and back-end generated)

Reply:
Ah ok gotcha all good then!